stakepooldclient/stakepooldclient.go

```diff
@@ -664,3 +663,68 @@ func (s *StakepooldManager) GetStakeInfo() (*pb.GetStakeInfoResponse, error) {
 	}
 	return nil, errors.New("GetStakeInfo RPC failed on all stakepoold instances")
 }
+
+// comaprePubkeys runs the gRPC command WalletMasterPubKeys on all stakepoold
+// instances, performs fn on all key, value pairs of received maps, and returns
+// whether any are true. Returns an error if any gRPC commands fail.
+func (s *StakepooldManager) comparePubKeys(fn func(string, string) bool) (bool, error) {
+	for i, conn := range s.grpcConnections {
+		client := pb.NewStakepooldServiceClient(conn)
+		resp, err := client.WalletMasterPubKeys(context.Background(), &pb.WalletMasterPubKeysRequest{})
+		if err != nil {
+			return false, fmt.Errorf("GetMasterPubkeys gRPC failed on stakepoold instance %d: %v", i, err)
+		}
+		for _, v := range resp.MasterPubKeys {
+			if fn(v.Account, v.PubKey) {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+// WalletsHavePubKey sends gRPC commands to all stakepoold and returns whether
+// pubkey exists on any wallets. Returns an error if any gRPC commands fail.
+func (s *StakepooldManager) WalletsHavePubKey(pubkey string) (bool, error) {
```

Comment:
I see that this is only used to confirm that the `coldwalletextpub` does not belong to the voting wallets. I think this check can be done at the same time as the master pub key for the default account is read to prevent performing the stakepoold `WalletMasterPubKeys` grpc request more than once.

1 idea is to invoke that grpc method once for ALL instances, ensuring that all the stakepoold instances return the same result (equal number of account-key pairs, with the same key value for each account on each backend server). Then the vetted account-key map is parsed to retrieve the default account pub key and also checked to ensure it does not contain the `coldwalletextpub`.

Reply:
I would rather keep the functions separate, for easier testing and such, but you are right that we only need to call the grpc once.  Because this operation only ever happens on start-up, I didn't think it was necessary, but we could cache the results, or use a clever closure scheme.

Reply:
If it's all the same, I've already had numerous reviews on this pr, and it's working ok.  Unless you think the gRPC commands will need to change, I'd rather further changes be made in another pr. That alright @itswisdomagain ?

Reply:
Sure, haven't tested myself but if others have tested and it works ok, further changes can be made in a subsequent PR.